File: fix_templates.py
"""
Run this script ONCE from C:\project_IFOA2\ to fix the template files.
It replaces the wrong orange templates with the correct green one.

Usage:  python fix_templates.py
"""
import shutil, os, sys

here = os.path.dirname(os.path.abspath(__file__))
green_src = os.path.join(here, 'recurrent_training_with_modules.pdf')

if not os.path.exists(green_src):
    print("ERROR: recurrent_training_with_modules.pdf not found in", here)
    sys.exit(1)

targets = ['Dispatch_graduate.pdf', 'HumanFactors.pdf']

for t in targets:
    dst = os.path.join(here, t)
    # Back up the wrong (orange) file first
    backup = dst.replace('.pdf', '_orange.pdf')
    if os.path.exists(dst):
        shutil.copy2(dst, backup)
        print(f"  Backed up  {t}  ->  {os.path.basename(backup)}")
    shutil.copy2(green_src, dst)
    print(f"  Replaced   {t}  with green template")

print("\nDone! Restart your backend server now.")

File: fix_templates.js
// Run once: node fix_templates.js
// Copies the correct green recurrent template as Dispatch_graduate.pdf and HumanFactors.pdf
const fs = require('fs');
const path = require('path');

const here = __dirname;
const green = path.join(here, 'recurrent_training_with_modules.pdf');

if (!fs.existsSync(green)) {
  console.error('ERROR: recurrent_training_with_modules.pdf not found');
  process.exit(1);
}

['Dispatch_graduate.pdf', 'HumanFactors.pdf'].forEach(name => {
  const dst = path.join(here, name);
  fs.copyFileSync(green, dst);
  console.log(`Created: ${name} (green template)`);
});

// Also create recurrent_training_orange.pdf from the HumanFactors orange
const hfOrange = path.join(here, 'HumanFactors_orange.pdf');
const recOrange = path.join(here, 'recurrent_training_orange.pdf');
if (fs.existsSync(hfOrange) && !fs.existsSync(recOrange)) {
  fs.copyFileSync(hfOrange, recOrange);
  console.log('Created: recurrent_training_orange.pdf');
}

console.log('\nDone! Now restart your backend server.');
